Model/Service/Category/CategoryMappingService.php

```
@@ -0,0 +1,140 @@
+     * @param NostoHelperData $nostoHelperData
+     * @param Logger $logger
+     */
+    public function __construct(
+        CollectionFactory $collectionFactory,
+        CategoryBuilder $categoryBuilder,
+        NostoHelperData $nostoHelperData,
+        Logger $logger
+    ) {
+        $this->collectionFactory = $collectionFactory;
+        $this->categoryBuilder = $categoryBuilder;
+        $this->nostoHelperData = $nostoHelperData;
+        $this->logger = $logger;
+    }
+
+    /**
+     * @param Store $store
+     * @return string
+     */
+    public function getCategoryMapping(Store $store): string
+    {
+        $array = $this->getMagentoCategories($store);
+        $mapping = json_encode((object)$array, JSON_UNESCAPED_SLASHES);
+        if ($mapping == false) {
+            return '';
+        }
+        return $mapping;
+    }
+
+    /**
+     * @param Store $store
+     * @return array
+     * @suppress PhanTypeMismatchArgument
+     */
+    private function getMagentoCategories(Store $store)
+    {
+
+        $categoriesArray = [];
+
+        try {
+            $categories = $this->collectionFactory->create()
+                ->addAttributeToSelect('*')
+                ->addIsActiveFilter()
+                ->setStore($store);
+
+            if (!$this->nostoHelperData->isAllCategoriesMapEnabled($store)) {
+                $categories->addAttributeToFilter('include_in_menu', ['eq' => 1]);
+            }
+
+            /** @var Category $category $item */
+            foreach ($categories->getItems() as $category) {
+                $hashedCategoryString = $this->hashCategoryString(strtolower(
+                    $this->categoryBuilder->getCategory($category, $store)
+                ));
+                if ($hashedCategoryString) {
+                    $categoriesArray[$hashedCategoryString] = $category->getUrl();
+                }
+            }
+        } catch (Exception $e) {
+            $this->logger->exception($e);
```

Do we really need to log this? If this breaks, the page will load but we will get a support ticket about something else. Let this bubble up.

---

It would still get caught by Magento, so it would not change much.